src/methods/join.js

```diff
@@ -445,10 +458,29 @@ export class JoinImplementation {
     let empty_effects = construct_empty_effects(realm);
     let v = realm.intrinsics.empty;
     if (c.consequent instanceof ReturnCompletion) {
-      let pnc = new PossiblyNormalCompletion(v, c.joinCondition, v, empty_effects, c.alternate, c.alternateEffects);
+      let pathCondition = [(AbstractValue.createFromUnaryOp(realm, "!", c.joinCondition): any)];
```

Comment:
Casting to any is generally not OK and this case is no exception. First assign the result of createFromUnaryOp to a local and add it to path only if it is in fact an AbstractValue. It does not seem correct for the result to be something other than an AbstractValue, so you may want to add an invariant to that effect.